lib/frames.rb

```diff
@@ -0,0 +1,77 @@
+require './lib/score_card'
+require './lib/roll'
+
+class Frames
+
+  def initialize
+    @round = 1
+    @score_count = 0
+    @strike_count = 0
+    @bonus = nil
+    @first_roll = true
+    @this_frame_count = 0
+  end
+
+  def pins_knocked
+    puts "How many pins did you knock down"
+    pins = gets.chomp.to_i
+  end
+
+
+  def new_round
+    if @round >= 11
+      ScoreCard.new(@score_count, @strike_count)
+    else
+      new_roll
+    end
+  end
+
+  def new_roll
+    @roll = Roll.new(score_count: @score_count, strike_count: @strike_count, first_roll: @first_roll, this_frame_count: @this_frame_count)
+    @roll.scoring(pins_knocked)
+    @bonus = @roll.strike_or_spare
+    roll_again?
+  end
+
+  def roll_again?
+    @roll.next_roll?
+    if @roll.roll_again == true
```

Comment:
Where possible, it's better to use return statements to communicate, rather than setting and reading instance variables. For example, this could read

```ruby
  roll_again = @roll.next_roll?
  if roll_again 
    @first_roll = false
    new_roll
  else ...
```

Ideally, you want to be minimising the state of a class as much as possible, and using return values instead.

Reply:
Can you spot any other places where you could return a value instead of storing it in an instance variable?